testing/jormungandr-integration-tests/src/mock/server.rs

```diff
@@ -189,146 +176,132 @@ impl JormungandrServerImpl {
     }
 }
 
+#[tonic::async_trait]
 impl Node for JormungandrServerImpl {
-    fn handshake(
+    type PullBlocksToTipStream = mpsc::Receiver<Result<Block, Status>>;
+    type GetBlocksStream = mpsc::Receiver<Result<Block, Status>>;
+    type PullHeadersStream = mpsc::Receiver<Result<Header, Status>>;
+    type GetHeadersStream = mpsc::Receiver<Result<Header, Status>>;
+    type GetFragmentsStream = mpsc::Receiver<Result<Fragment, Status>>;
+    type BlockSubscriptionStream =
+        Pin<Box<dyn Stream<Item = Result<BlockEvent, Status>> + Send + Sync + 'static>>;
```

Comment:
You can use [BoxFuture](https://docs.rs/futures/0.3.5/futures/future/type.BoxFuture.html) as a shorthand.

Reply:
Strange, I tried to use it:

```
 type BlockSubscriptionStream = BoxFuture<'static, Result<BlockEvent, Status>>;
```

but got compilation issue:

```
error[E0277]: the trait bound `(dyn core::future::future::Future<Output = std::result::Result<mock::server::node::BlockEvent, tonic::status::Status>> + std::marker::Send + 'static): futures_core::stream::Stream` is not satisfied
   --> testing\jormungandr-integration-tests\src\mock\server.rs:173:6
    |
173 | impl Node for JormungandrServerImpl {
    |      ^^^^ the trait `futures_core::stream::Stream` is not implemented for `(dyn core::future::future::Future<Output = std::result::Result<mock::server::node::BlockEvent, tonic::status::Status>> + std::marker::Send + 'static)`
    |
    = note: required because of the requirements on the impl of `futures_core::stream::Stream` for `std::pin::Pin<std::boxed::Box<(dyn core::future::future::Future<Output = std::result::Result<mock::server::node::BlockEvent, tonic::status::Status>> + std::marker::Send + 'static)>>`

error[E0277]: `(dyn core::future::future::Future<Output = std::result::Result<mock::server::node::BlockEvent, tonic::status::Status>> + std::marker::Send + 'static)` cannot be shared between threads safely
   --> testing\jormungandr-integration-tests\src\mock\server.rs:179:36
    |
179 |     type BlockSubscriptionStream = BoxFuture<'static, Result<BlockEvent, Status>>;
    |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `(dyn core::future::future::Future<Output = std::result::Result<mock::server::node::BlockEvent, tonic::status::Status>> + std::marker::Send + 'static)` cannot be shared between threads safely
```

Reply:
Oops, I should have seen it's a stream. Use `BoxStream` for streams.

Reply:
It might be that `BoxStream` is still not enough because `Sync` is also required.
In chain-network, I used a custom generic type with a `Stream` impl because it has to convert things. If that's too much to do, leave this as is, `BoxStream` is only cosmetics.